packages/sitecore-jss-angular/src/services/models.ts

```diff
@@ -0,0 +1,17 @@
+import { LayoutServiceContextData, RouteData } from '../public_api';
+
+export class JssState {
```

Comment:
Some definitions, like "LayoutServiceError", "serverRoute" should be defined in the app instead, since it fully depends on customers' implementation.
Here are a few questions for you to think about:
1. As a developer, If I'm using a different _LayoutServiceError_, how can I bypass the type that is defined here?
1. As a developer, If I'm using a custom JssState and some properties are not needed for me OR I want to set additional properties to the state, how can I integrate a new solution?
1. Think about existing customers and the customizations that they can have in their state

Reply:
All of this can be achieved with Angular's dependency injection, but you're right, it will be a pain.
I've modified the approach to still have the connection between packages state and app state, but JssState is back into sample